packages/frontpage/lib/schema.ts

```diff
@@ -24,6 +24,8 @@ export const PostVote = pgTable("post_votes", {
     .references(() => Post.id),
   createdAt: timestamp("created_at").notNull().defaultNow(),
   authorDid: text("author_did").notNull(),
+  cid: text("cid").notNull().unique(),
+  rkey: text("rkey").notNull().unique(),
```

Comment:
After I made this change I realised that rkeys aren't actually guaranteed to be unique.

See https://atproto.com/specs/record-key

This isn't a problem unless someone does something dodgy, but the issue is right now we're relying on it for deduplication of votes and also for handling deleting of posts, comments, and votes...

I don't think this causes any massive issues right now but may do in the future if we expand our reliance on this constraint. So we should move to something else (varies per use case), but can be a future refactor.

Reply:
Now I realised this PR doesn't handle duplicate votes from the same did 🤦 will fix before merging this PR but will likely leave that bigger refactor to later.

Reply:
8eb759def9deb2d458807acc071f411e5eaf1cf2 includes the fix, https://linear.app/likeandscribe/issue/UN-67/remove-and-replace-unique-rkey-constraints is the ticket for the refactor.